src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -82,6 +82,39 @@ class RsErrorAnnotator : RsAnnotatorBase(), HighlightRangeExtension {
         element.accept(visitor)
     }
 
+    private fun checkEnumItem(holder: AnnotationHolder, o: RsEnumItem) {
+        checkDuplicates(holder, o)
+        o.enumBody?.let { checkDuplicateEnumVariants(holder, it) }
+    }
+
+    private fun checkDuplicateEnumVariants(holder: AnnotationHolder, o: RsEnumBody) {
+        var discrCounter = 0
+        val indexToVariantMap = mutableMapOf<Int, Pair<RsEnumVariant, /* already reported */Boolean>>()
```

Comment:
Let's replace `Pair<RsEnumVariant, Boolean>` with some data class, for example:
```kotlin
private data class VariantInfo(val variant: RsEnumVariant, val alreadyReported: Boolean)
```
After some refactoring:
```kotlin
private fun checkDuplicateEnumVariants(holder: AnnotationHolder, o: RsEnumBody) {
    var discrCounter = 0
    val indexToVariantMap = hashMapOf<Int, VariantInfo>()
    for (variant in o.enumVariantList) {
        val literal = variant.variantDiscriminant?.expr as? RsLitExpr
        val int = literal?.integerLiteralValue?.toIntOrNull()
        val idx = int ?: discrCounter
        discrCounter = idx + 1

        val previous = indexToVariantMap[idx]
        if (previous != null) {
            RsDiagnostic.DuplicateEnumDiscriminant(variant, idx).addToHolder(holder)
            if (!previous.alreadyReported) {
                RsDiagnostic.DuplicateEnumDiscriminant(previous.variant, idx).addToHolder(holder)
                indexToVariantMap[idx] = previous.copy(alreadyReported = true)
            }
        } else {
            indexToVariantMap[idx] = VariantInfo(variant, alreadyReported = false)
        }
    }
}
```

Reply:
Thanks! It does look better now